Do not do any queries if there were no migrations before.

Ignas opened the pull request:
## Summary

This works around java.util.NoSuchElementException being returned by cassandra
server when executing an empty batch.

<hr>

## Pull Request (PR) Checklist

### Documentation
- [x] Documentation in `README.md` or [Wiki](https://github.com/builtamont-oss/cassandra-migration/wiki) updated
- [x] Update [Release Notes](https://github.com/builtamont-oss/cassandra-migration/releases) if applicable -- collaborator-access only

### Code Review
- [x] Self code review -- take another pass through the changes yourself

### Tests
- [x] All tests passes

hhandoko commented:
Thanks for your contribution @Ignas ! Before I can merge it, can you please write some test to check this condition?

hhandoko commented on src/main/java/com/builtamont/cassandra/migration/internal/dbsupport/SchemaVersionDAO.kt:
```diff
@@ -309,6 +309,10 @@ open class SchemaVersionDAO(private val session: Session, val keyspaceConfig: Ke
+        if (migrationVersions.size == 0) {
+            return 1;
+        }
+
```
This could be rewritten as `if (migrationVersions.isEmpty()) return 1` guard clause.

Ignas replied:
Doesn't this line prevent batch from being executed with statements `return i + 1`?

hhandoko commented:
Hi @Ignas , can you elaborate the steps to reproduce the issue (i.e. triggering the `NoSuchElementException`)?

I've tried adding an empty migrations directory and no exceptions are thrown. The for-loop at `for (versionRow in versionRows) { ... }` and `for (i in migrationVersions.indices) { ... }` are skipped if no prior migrations exists or `versionRows` is empty.

Ignas commented:
Issue #65 is resolved by the change. This can be easily reproduced with newer cassandra drivers.

Ignas commented:
Ok, strike that, this does not resolve the issue, it seems that even when versions table is populated empty batch still gets executed. So the fix should instead check that batch is empty and skip it.

Ignas committed:
Removed dead code.

This works around java.util.NoSuchElementException being returned by cassandra
server when executing an empty batch.

Ignas commented:
I just removed the batch generation code, as if batch has statements in it - it does not get executed ever.

hhandoko commented:
Thank you for your PR, sorry it was left open for too long!
